chia/full_node/mempool_manager.py

```diff
@@ -157,61 +222,36 @@ async def create_bundle_from_mempool(
-                    item_cost = item.cost - cost_saving
-                    log.info(f"Cumulative cost: {cost_sum}, fee per cost: {item.fee / item_cost}")
-                    if (
-                        item_cost + cost_sum <= self.limit_factor * self.constants.MAX_BLOCK_COST_CLVM
-                        and item.fee + fee_sum <= self.constants.MAX_COIN_AMOUNT
-                    ):
-                        # Update the deduplicated coin spends map
-                        dedup_coin_spends.update(new_dedup_coin_spends)
-                        # Add all the spends in the transaction except the duplicated ones
-                        for cs in item.spend_bundle.coin_spends:
-                            if cs not in spends_to_dedup:
-                                coin_spends.append(cs)
-                                removals.append(cs.coin)
-                        additions.update(item.additions)
-                        sigs.append(item.spend_bundle.aggregated_signature)
-                        cost_sum += item_cost
-                        fee_sum += item.fee
-                        processed_spend_bundles += 1
-                    else:
-                        broke_from_inner_loop = True
-                        break
+                result = check_item_for_dedup(item, dedup_coin_spends)
+                if result is None:
+                    continue
+                spends_to_dedup, cost_saving, new_dedups = result
+                item_cost = item.cost - cost_saving
+                log.info(f"Cumulative cost: {cost_sum}, fee per cost: {item.fee / item_cost}")
+                if (
+                    item_cost + cost_sum <= self.limit_factor * self.constants.MAX_BLOCK_COST_CLVM
+                    and item.fee + fee_sum <= self.constants.MAX_COIN_AMOUNT
+                ):
+                    # Update the deduplicated coin spends map
+                    dedup_coin_spends.update(new_dedups)
+                    # Add all the spends in the transaction except the duplicated ones
+                    for cs in item.spend_bundle.coin_spends:
+                        if cs not in spends_to_dedup:
+                            coin_spends.append(cs)
+                            removals.append(cs.coin)
+                    # The assumption here is that deduplicated spends generate
+                    # the same additions (as a result of spending the coin with
+                    # the same solution), so those additions get deduplicated
+                    # naturally on set update.
+                    # Ideally we would have more explicit checks
+                    additions.update(item.additions)
+                    sigs.append(item.spend_bundle.aggregated_signature)
+                    cost_sum += item_cost
+                    fee_sum += item.fee
+                    processed_spend_bundles += 1
+                else:
+                    broke_from_inner_loop = True
```

Comment:
when I see this patter, of recording that we need to break another level of loop, I think that the code would probably benefit from being pulled out into a function that can just `return` out of the two loops instead,